submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/parser/ExperimentSpecParser.java

```diff
@@ -168,8 +171,35 @@ private static V1PodTemplateSpec parseTemplateSpec(
     container.setResources(resources);
     container.setEnv(parseEnvVars(taskSpec, experimentSpec.getMeta().getEnvVars()));
     containers.add(container);
-    podSpec.setContainers(containers);
 
+    /**
+     * Init Git localize Container
+     */
+    if (experimentSpec.getCode() != null) {
+      CodeLocalizer localizer = AbstractCodeLocalizer.getCodeLocalizer(
+          experimentSpec.getCode().getSyncMode(),
+          experimentSpec.getCode().getUrl());
+      localizer.localize(podSpec);
+
+      if (podSpec.getInitContainers() != null
+          && podSpec.getInitContainers().size() > 0) {
+        String volumeName = podSpec.getInitContainers().get(0).getVolumeMounts()
```

Comment:
Is this part of logic generic enough to support multiple volumes? (It is possible in the future we want to support mount other volumes to the pod, such as an NFS volume. Instead of get(0), can we get the volume by dir name?

Reply:
(Yes, pod containers may mount multiple volumes in future for various reasons which can be pat of other init container process). But at any point of time, this "code localizer" init container would have only 1 volume mount. However, open to change if you think otherwise.

Reply:
It makes sense to me. We can revisit this once we want to support multiple volumes.